pkg/chunk/storage/factory.go

```diff
@@ -48,10 +55,21 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	cfg.BoltDBConfig.RegisterFlags(f)
 	cfg.FSConfig.RegisterFlags(f)
 
+	f.StringVar(&cfg.Engine, "store.engine", "chunks", "The storage engine to use: chunks or tsdb. Be aware tsdb is experimental and shouldn't be used in production.")
 	cfg.IndexQueriesCacheConfig.RegisterFlagsWithPrefix("store.index-cache-read.", "Cache config for index entry reading. ", f)
 	f.DurationVar(&cfg.IndexCacheValidity, "store.index-cache-validity", 5*time.Minute, "Cache validity for active index entries. Should be no higher than -ingester.max-chunk-idle.")
 }
 
+// Validate config and returns error on failure
+// TODO need to rebase to call it, because Config validation has been added just recently
```

Comment:
Do you mind rebasing, so we can get this done as well? Once rebased, the following should be added to the `Validate()` function in `pkg/cortex/cortex.go`:

```
if err := c.Storage.Validate(); err != nil {
		return errors.Wrap(err, "invalid storage config")
	}
	if err := c.TSDB.Validate(); err != nil {
		return errors.Wrap(err, "invalid TSDB config")
	}
```

Reply:
done